ci/test/lint-main/checks/check-cargo.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
+# lint — complains about misformatted files and other problems.
+
+set -euo pipefail
+
+cd "$(dirname "$0")/../../../.."
+
+. misc/shlib/shlib.bash
+
+try bin/lint-cargo
+
+try_status_report
```

Comment:
This is now called several times and it prints outputs like this:

```
^^^ 🚨 Failed: xargs shellcheck -P SCRIPTDIR
0/1 commands passed
```
where "1" refers to the number of `try` in `check-shell-files.sh` and 0 refers to the fact that none failed. Consider extending the macro if possible so that it is a bit more verbose


```
0/1 commands passed in check-shell-files.sh
```

Or something along those lines, if it does not take more than 5 min.

Reply:
The error is in the section `check-shell-files.sh`. Furthermore, I only print output by default if a check fails.

```
+++ check-shell-files.sh (FAILED)
In ci/test/lint-main/checks/check-shell-files.sh line 23:
if [[ "$$(uname)" == "Linux" ]]
                  ^-- SC2193 (warning): The arguments to this comparison can never be equal. Make sure your syntax is correct.

For more information:
  https://www.shellcheck.net/wiki/SC2193 -- The arguments to this comparison ...
^^^ 🚨 Failed: xargs shellcheck -P SCRIPTDIR
0/1 commands passed
```

I am happy to revisit this in future if it turns out to be confusing.